R/plot_star_system.R

```diff
@@ -38,76 +40,100 @@
-    assert_choice(spectral_type, choices = c("O", "B", "A", "F", "G", "K", "M"))
-  }
+  assert_character(spectral_type, len = 1, any.missing = FALSE)
+  assert_choice(spectral_type, choices = c(" ", "O", "B", "A", "F", "G", "K", "M"))
+
+  star_name = paste0(
+    "Central star",
+    if (!is.null(spectral_type)) paste0(", type ", spectral_type) else ""
+  )
 
-  # Create data frame for plotting the star system:
-  # First row is always the host star and following rows are the planets.
-  #    - randomize position on the orbit, with the star being in the center (0)
-  #    - rescale orbit for the visualization, keep the star at center (0). Add habitable zone edges.
-  #    - ensure star is the largest object with size 17
-  #    - map color based on spectral type or density
   plot_data = data.frame(
     orbit_offset = c(0, runif(nrow(planet_data), min = 0, max = 2 * pi), 0, 0),
     pl_orbsmax = c(0, .rescale_orbsmax(c(
       planet_data$pl_orbsmax,
-      min(habitable_zone), max(habitable_zone)
+      min(habitable_zone),
+      max(habitable_zone)
     ))),
-    pl_rade = c(17, planet_data$pl_rade, 0, 0), # Ensure star is the largest object
-    col = c(.map_star_color(spectral_type), .map_planet_color(planet_data$pl_dens), "", "")
+    pl_rade = c(17, planet_data$pl_rade, 0, 0),
+    type = c(spectral_type, .map_planet_type(planet_data$pl_dens), "HZ inner edge", "HZ outer edge")
   )
 
-  inner_hz = plot_data$pl_orbsmax[length(plot_data$pl_orbsmax) - 1]
-  outer_hz = plot_data$pl_orbsmax[length(plot_data$pl_orbsmax)]
-
+  inner_hz = plot_data$pl_orbsmax[nrow(plot_data) - 1]
+  outer_hz = plot_data$pl_orbsmax[nrow(plot_data)]
   plot_data = plot_data[1:(nrow(plot_data) - 2), ]
 
+  color_map = c(
+    " " = "white",
+    "M" = "red2",
+    "K" = "indianred2",
+    "G" = "yellow1",
+    "F" = "lightyellow",
+    "A" = "lavenderblush",
+    "B" = "lightsteelblue1",
+    "O" = "slateblue1",
+    "Gas-dominated planet" = "lightblue",
+    "Water/ice-rich planet" = "deepskyblue",
+    "Terrestial planet" = "darkorange3",
+    "Iron-rich planet" = "ivory3",
+    "Super-dense planet" = "gray20"
+  )
+
+  names(color_map)[1] = paste0("Central star, type: ", spectral_type)
 
-  ggplot(plot_data, aes(x = orbit_offset, y = pl_orbsmax, size = pl_rade, color = col)) +
+  ggplot(plot_data, aes(x = orbit_offset, y = pl_orbsmax, size = pl_rade, color = type)) +
     annotate(
       "rect",
-      xmin = -Inf, xmax = Inf,  # full width
+      xmin = -Inf, xmax = Inf,
       ymin = inner_hz, ymax = outer_hz,
       alpha = 0.15, fill = "green"
     ) +
     geom_hline(yintercept = plot_data$pl_orbsmax, color = "grey15") +
     geom_point() +
-    scale_color_identity() +
-    scale_size_continuous(range = c(3.5, 17)) + # Limit object size to prevent it from overlapping
+    scale_color_manual(
+      values = color_map,
+      labels = setNames(
+        paste0("Central star, spectral type ", spectral_type),
```

Comment:
issue(non-blocking): You create `star_name` variable, but do not use it anywhere. I think it could go here.
```suggestion
        star_name
```